components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -2073,12 +713,12 @@ pub fn parse_color(string: &str) -> Result<RGBA, ()> {
 
 // Used by drawImage to determine if a source or destination rectangle is valid
 // Origin coordinates and size cannot be negative. Size has to be greater than zero
-fn is_rect_valid(rect: Rect<f64>) -> bool {
+pub fn is_rect_valid(rect: Rect<f64>) -> bool {
```

Comment:
Let's move these helper functions to the new file instead of making them public.